.github/workflows/workflow_pr.yml

```diff
@@ -70,9 +70,30 @@ jobs:
         run: |
           bun run utils/bump-release-version.ts check
 
+  pre_commit:
+    name: Verify pre-commit hooks pass
+    runs-on: public-amd64
+    if: ${{ contains(github.event.pull_request.labels.*.name, 'ready') }}
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
```

Comment:
you can make this much faster and less code by using `uv`

```yaml
- name: Setup uv
  uses: astral-sh/setup-uv@v6

- name: Run pre-commit
  uses: uvx pre-commit run --all-files --show-diff-on-failure --color always
```

caching should work ootb